AX: Merge AXImage into AccessibilityRenderObject

hoffmanjoshua opened the pull request:
#### 300dc2680bab73a44d56758df86cff3252b47105
<pre>
AX: Merge AXImage into AccessibilityRenderObject
<a href="https://bugs.webkit.org/show_bug.cgi?id=296742">https://bugs.webkit.org/show_bug.cgi?id=296742</a>
<a href="https://rdar.apple.com/157215308">rdar://157215308</a>

Reviewed by Tyler Wilcock.

Eliminate the need for AXImage by moving its logic into AccessibilityRenderObject.

This also fixes a unified sources build issues in AccessibilityTableColumn (importing
AXObjectCache).

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/accessibility/AXImage.cpp: Removed.
* Source/WebCore/accessibility/AXImage.h: Removed.
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::toSimpleImage):
(WebCore::AXObjectCache::createObjectFromRenderer):
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::associatedImageObject const):
(WebCore::AccessibilityRenderObject::accessibilityImageMapHitTest const):
(WebCore::AccessibilityRenderObject::imageOverlayElements):
(WebCore::AccessibilityRenderObject::associatedAXImage const): Deleted.
* Source/WebCore/accessibility/AccessibilityRenderObject.h:
* Source/WebCore/accessibility/AccessibilityTableColumn.cpp:

Canonical link: <a href="https://commits.webkit.org/298159@main">https://commits.webkit.org/298159@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6e0b68e9b5a415b3d26af1ae7e574d47c20185ee

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114245 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33992 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/24454 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/120411 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64980 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116134 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34621 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42553 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86820 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41762 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117193 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27571 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102608 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67210 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26752 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20734 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64103 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96945 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20851 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123627 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41263 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30770 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95655 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41639 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98810 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95438 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24356 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40576 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18407 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/37343 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41143 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/46652 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40746 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44052 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42497 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8164540bc2c18e2461f3aae2bf646b67ad476e74. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48775

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5378179d9c8b10c84ec32c9e95b7dbc1914fbce0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48775

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2b92bb94a58c8a9a0cd0c597acf02725885dd567. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48775

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6e0b68e9b5a415b3d26af1ae7e574d47c20185ee. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48775

hoffmanjoshua committed:
AX: Merge AXImage into AccessibilityRenderObject
https://bugs.webkit.org/show_bug.cgi?id=296742
rdar://157215308

Reviewed by Tyler Wilcock.

Eliminate the need for AXImage by moving its logic into AccessibilityRenderObject.

This also fixes a unified sources build issues in AccessibilityTableColumn (importing
AXObjectCache).

* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/accessibility/AXImage.cpp: Removed.
* Source/WebCore/accessibility/AXImage.h: Removed.
* Source/WebCore/accessibility/AXObjectCache.cpp:
(WebCore::toSimpleImage):
(WebCore::AXObjectCache::createObjectFromRenderer):
* Source/WebCore/accessibility/AXObjectCache.h:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::associatedImageObject const):
(WebCore::AccessibilityRenderObject::accessibilityImageMapHitTest const):
(WebCore::AccessibilityRenderObject::imageOverlayElements):
(WebCore::AccessibilityRenderObject::associatedAXImage const): Deleted.
* Source/WebCore/accessibility/AccessibilityRenderObject.h:
* Source/WebCore/accessibility/AccessibilityTableColumn.cpp:

Canonical link: https://commits.webkit.org/298159@main
